test/components/views/rooms/wysiwyg_composer/utils/createMessageContent-test.ts

```diff
@@ -41,117 +41,146 @@ describe("createMessageContent", () => {
         });
-    });
 
-    it("Should strip the /me prefix from a message", async () => {
-        const textBody = "some body text";
-        const content = await createMessageContent(EMOTE_PREFIX + textBody, true, { permalinkCreator });
+        it("Should strip single / from message prefixed with //", async () => {
+            const content = await createMessageContent("//twoSlashes", true, { permalinkCreator });
 
-        expect(content).toMatchObject({ body: textBody, formatted_body: textBody });
-    });
+            expect(content).toMatchObject({ body: "/twoSlashes", formatted_body: "/twoSlashes" });
+        });
 
-    it("Should strip single / from message prefixed with //", async () => {
-        const content = await createMessageContent("//twoSlashes", true, { permalinkCreator });
+        it("Should set the content type to MsgType.Emote when /me prefix is used", async () => {
+            const textBody = "some body text";
+            const content = await createMessageContent(EMOTE_PREFIX + textBody, true, { permalinkCreator });
 
-        expect(content).toMatchObject({ body: "/twoSlashes", formatted_body: "/twoSlashes" });
+            expect(content).toMatchObject({ msgtype: MsgType.Emote });
+        });
     });
 
-    it("Should set the content type to MsgType.Emote when /me prefix is used", async () => {
-        const textBody = "some body text";
-        const content = await createMessageContent(EMOTE_PREFIX + textBody, true, { permalinkCreator });
+    describe("Plaintext composer input", () => {
+        it("Should replace at-room mentions with `@room` in body", async () => {
+            const messageComposerState = `<a href="#" contenteditable="false" data-mention-type="at-room" style="some styling">@room</a> `;
+
+            const content = await createMessageContent(messageComposerState, false, { permalinkCreator });
+            expect(content).toMatchObject({ body: "@room " });
+        });
+
+        it("Should replace user mentions with user name in body", async () => {
+            const messageComposerState = `<a href="https://matrix.to/#/@test_user:element.io" contenteditable="false" data-mention-type="user" style="some styling">a test user</a> `;
+
+            const content = await createMessageContent(messageComposerState, false, { permalinkCreator });
 
-        expect(content).toMatchObject({ msgtype: MsgType.Emote });
+            expect(content).toMatchObject({ body: "a test user " });
+        });
+
+        it("Should replace room mentions with room mxid in body", async () => {
+            const messageComposerState = `<a href="https://matrix.to/#/#test_room:element.io" contenteditable="false" data-mention-type="room" style="some styling">a test room</a> `;
+
+            const content = await createMessageContent(messageComposerState, false, { permalinkCreator });
+
+            expect(content).toMatchObject({
+                body: "#test_room:element.io ",
+            });
+        });
```

Comment:
These are the new tests, all other changes in this file result from increased nesting